### PR TITLE
Attempt to make use of (sub)package/(sub)module consistent

### LINKED ISSUE #47: figure out the right wording for (sub)package & (sub)module)
We now use subpackage, package, module and submodule very inconsistently in various parts of the paper. Let's go through once we have completed most/all sections and do this right.

### PR DESCRIPTION
**Pull request by mdhaber**
Key word being "attempt". I'm not sure if all of these changes are correct, but this will hopefully at least highlight potential misuses of the these words.
Closes #47

### CODE REVIEW

File: scipy-1.0/subpackages.tex [hunk 1]
```diff
@@ -47,20 +45,20 @@
...
-    This package contains various functions for multi-dimensional image
+    This subpackage contains various functions for multi-dimensional image
     processing, including convolution and assorted linear and nonlinear
     filters (Gaussian filter, median filter, Sobel filter, etc.);
     interpolation; region labeling and processing; and mathematical morphology
     functions.
 \item[\texttt{misc}]
     A collection of functions that did not fit into the other subpackages.
     While this subpackage still exists in SciPy 1.0, effort is underway
-    to deprecate or relocate the contents of this subpackage and eventually remove it.
```
Comment (mdhaber) [inline scipy-1.0/subpackages.tex]: I removed this word purely for formatting: "it" was by itself on the next line.  I don't think this is inaccurate. Work is underway to deprecate/relocate its contents and remove it.

File: scipy-1.0/paper.tex [hunk 2]
```diff
@@ -872,8 +872,8 @@ \subsection*{Cython bindings for BLAS, LAPACK, and special}
 LAPACK\cite{LAPACK} routines for many years. SciPy now additionally
 includes Cython\cite{behnel2011cython} wrappers for
 many BLAS and LAPACK routines (added in 2015) and the special functions 
-provided in the \texttt{scipy.{\allowbreak}special} submodule (added in 2016).  
-These Cython wrappers are available in the modules
+provided in the \texttt{scipy.{\allowbreak}special} subpackage (added in 2016).
```
Comment (tylerjereddy) [inline scipy-1.0/paper.tex]: That looks ok I think:

```
In [8]: help(special)
Help on package scipy.special in scipy:

NAME
    scipy.special
```

But the official docs so allow use of (sub) prefix for these of course.

File: scipy-1.0/ckdtree.tex [hunk 3]
```diff
@@ -1,4 +1,4 @@
-The \texttt{cKDTree} module, which implements a space-partitioning data structure that
+The \texttt{scipy.spatial.cKDTree} subpackage, which implements a space-partitioning data structure that
```
[the remark this answers is not in the document]
Comment (mdhaber) [reply scipy-1.0/ckdtree.tex]: According to the interpreter, `scipy` is a module, too.
```
import scipy
print(repr(scipy))

```
But that is not consistent with the definition of a module provided in the [documentation](https://docs.python.org/3/tutorial/modules.html). 

"A module is a file containing Python definitions and statements."

(@rgommers seemed to prefer this strict definition in #47 if I understood correctly.)

I checked before committing, and cKDTree isn't a file.

[the remark this answers is not in the document]
Comment (mdhaber) [reply scipy-1.0/ckdtree.tex]: But yes, I'll change the capitalization. Thanks for catching that.

File: scipy-1.0/subpackages.tex [hunk 4]
```diff
@@ -10,9 +10,7 @@
 \begin{description}[leftmargin=!, labelwidth=\widthof{\bfseries \texttt{interpolate}}]
 \itemsep0em
 \item[\texttt{cluster}]
-    The \texttt{cluster} subpackage contains two modules:
-    \texttt{cluster.vq} provides vector quantization and $k$-means algorithms,
-    and \texttt{cluster.hierarchy} provides functions for hierarchical and
+    The \texttt{cluster} subpackage contains \texttt{cluster.vq}, which provides vector quantization and $k$-means algorithms, and \texttt{cluster.hierarchy}, which provides functions for hierarchical and
```
Comment (mdhaber) [inline scipy-1.0/subpackages.tex]: These are indeed modules. Should I revert?

File: scipy-1.0/subpackages.tex [hunk 5]
```diff
@@ -73,23 +71,19 @@
...
+		decomposition and LU factorization.
     \texttt{scipy.sparse.csgraph} provides a collections of graph algorithms
     for which the graph is represented using a sparse matrix.  Algorithms
     include connected components, shortest path, minimum spanning tree
     and more.
 \item[\texttt{spatial}]
     This subpackage provides spatial data structures and algorithms,
     including the $k$-d tree, Delaunay triangulation, convex hulls and Voronoi
-    diagrams.  The subpackage \texttt{scipy.spatial.distance} provides
+    diagrams. \texttt{scipy.spatial.distance} provides
```
Comment (mdhaber) [inline scipy-1.0/subpackages.tex]: This is actually a module, too.

Comment (tylerjereddy) [reply scipy-1.0/subpackages.tex]: I think your decision to remove the specification here is just fine, so I'll mark as resolved.

File: scipy-1.0/subpackages.tex (hunk 4 above)
Comment (tylerjereddy) [reply scipy-1.0/subpackages.tex]: Nah, what you did looks fine to me. Completely avoiding the confusion seems nice too.

**Comment by tylerjereddy** [comment]
Ok, minor revisions were made--in it goes.

The Python [glossary](https://docs.python.org/3/glossary.html#term-package) makes the situation even more complicated to think about:

> package
> A Python module which can contain submodules or recursively, subpackages. Technically, a package is a Python module with an __path__ attribute.

Of course, SciPy compiles extension modules from Cython and so on, too. I suspect there could be a fairly long debate about some individual cases, but I'm happy for the issue to be considered closed! They are things available in the SciPy namespace!

The glossary also defines `namespace package` from PEP420, which serves only as a container for subpackages, but have no physical representation...

**Comment by tylerjereddy** [comment]
thanks Matt

**Pull request merged**